spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -103,4 +126,29 @@ public MessageChannel resolveDestination(String channelName) {
 			return channel;
 		}
 	}
+
+	/**
+	 * Configure a new destination before it is bound.
+	 * @param <T> the extended properties type. If you need to support dynamic binding
+	 * with multiple binders, use {@link Object} and cast as needed.
+	 *
+	 * @since 2.0
+	 *
+	 */
+	@FunctionalInterface
+	public interface NewBindingCallback<T> {
```

Comment:
Perhaps the name for this callback could be `DynamicDestinationBindingCallback` ?

Reply:
Sounds good; I'll hold off changing in case there are other suggestions 😄

Reply:
@garyrussell I agree with @ilayaperumalg on the name

Reply:
Actually I'd call it  `NewDestinationBindingCallback`. There is nothing dynamic about this interface's definition. Implementations on the other hand could have a more specific names.